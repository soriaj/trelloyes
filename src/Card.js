import React from 'react';
import './Card.css'

class Card extends React.Component{
   render(){
      const { title } = this.props
      const { content } = this.props
      return(
         <div className='Card'>
            <button 
               onClick={() => this.props.onDeleteItem(this.props.cardId)}
               type='button'>
               delete
               </button>
            <h3>{title}</h3>
            <p>{content}</p>
         </div>
      )
   }
}
// function Card(props){
//    // const title = props.title;
//    // const content = props.content;
//    return (
//       <div className='Card'>
//          <button type='button'>delete</button>
//          <h3>{props.title}</h3>
//          <p>{props.content}</p>
//       </div>
//    )
// }

export default Card;