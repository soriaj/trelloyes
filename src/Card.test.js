import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import Card from './Card';

describe('Cards component', () => {
   it('renders UI without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<Card />, div);
      ReactDOM.unmountComponentAtNode(div);
	});
	
	it('renders the UI as expected with Snapshot', () => {
		const tree = TestRenderer.create(<Card title='First list' content='Some Content for Testing' />).toJSON();
		expect(tree).toMatchSnapshot();
	});

});