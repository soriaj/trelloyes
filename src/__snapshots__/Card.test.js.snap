// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards component renders the UI as expected with Snapshot 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    First list
  </h3>
  <p>
    Some Content for Testing
  </p>
</div>
`;
